src/execution/processor.rs

```diff
@@ -0,0 +1,673 @@
+use super::root_hash;
+use crate::{
+    chain::{
+        dao,
+        intrinsic_gas::*,
+        protocol_param::{fee, param},
+        validity::ValidationError,
+    },
+    execution::evm,
+    models::*,
+    state::IntraBlockState,
+    State,
+};
+use ethereum_types::*;
+use evmodin::{Revision, StatusCode};
+use std::cmp::min;
+use TransactionAction;
+
+pub struct ExecutionProcessor<'storage, 'r, 'h, 'b, 'c, S>
```

Comment:
as far as I understand this is related to a single block?

maybe rename:
```suggestion
pub struct BlockExecutionProcessor<'storage, 'r, 'h, 'b, 'c, S>
```